#! /bin/sh -e

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

From: "John David Anglin" <dave@hiauly1.hia.nrc.ca>
Sender: gcc-patches-owner@gcc.gnu.org
To: gcc-patches@gcc.gnu.org, java-patches@gcc.gnu.org
Subject: [committed] Change default for enable_hash_synchronization_default on PA-RISC
Date: Tue, 13 Jun 2006 21:51:55 -0400 (EDT)

I've noticed that we get occassional failures of process intensive
tests in the libjava testsuite (e.g., Process_3).  Carlos O'Donell
indicated that hashing locks might be unsafe because of the special
way locks are handled in parisc-linux.  I've also seen a few failures
in the libjava suite on hpux.  So, I've decided that it would be
better to be conservative in this regard.

With the patch installed, I've seen no libjava testsuite failures
on hppa-unknown-linux-gnu on the trunk in several runs.

Committed to trunk.  If the patch works out, I will probably backport.

Dave
-- 
J. David Anglin                                  dave.anglin@nrc-cnrc.gc.ca
National Research Council of Canada              (613) 990-0752 (FAX: 952-6602)

# DP: 2006-06-13  John David Anglin  <dave.anglin@nrc-cnrc.gc.ca>
# DP: 
# DP: 	* configure.host (hppa*-*): Set 'enable_hash_synchronization_default' to
# DP: 	no.

Index: configure.host
===================================================================
--- libjava/configure.host	(revision 114624)
+++ libjava/configure.host	(working copy)
@@ -130,7 +130,7 @@
   hppa*-*)
 	sysdeps_dir=pa
 	libgcj_interpreter=yes
-	enable_hash_synchronization_default=yes
+	enable_hash_synchronization_default=no
 	;;
   powerpc64*-*)
 	sysdeps_dir=powerpc
